### DISCUSSION TITLE
WASM loader?

### DISCUSSION THREAD

**Discussion by sturlath** [Megabit/Blazorise#3292] [Ideas]
I'm at the end of my development so getting ready to launch my site in c.a. 2 weeks and am starting to get test feedback on slow loading of my site.

This simple boring spinner for 3-5 seconds is just not it...
![image](https://user-images.githubusercontent.com/2386572/148376457-b4b3fd92-1587-4655-8ef5-1d82c946aafb.png)

So I just had an idea for a super simple component that I think lots of backend WASM sites would love to have and that is some kind of loading icon/image.

These messages shown (one after another) when the site is loading

> We are loading your assets...

> Assets loaded...

> Importing your stuff...

It wouldn´t be doing any of these things but the user probably has more patience when I "knows" why he is waiting so long.

The component would have steps (e.g. 3 as here above) with text input  and some basic animation (icons/colours, time between steps?). 

Is this even possible? I know that there are some other ways to [pre-render ](https://docs.microsoft.com/en-us/aspnet/core/blazor/components/prerendering-and-integration?view=aspnetcore-6.0&pivots=webassembly) stuff on the server and I probably will do that but thats looks to be lot more work for me, with time I really don´t have.

Thought I share this with you if this was something..

**Comment by stsrki** [comment]
Well, we already have it :)

https://blazorise.com/docs/extensions/spinkit

**Reply by sturlath** [reply]
Yes I knew about that one and I think that is just as "bad" as the one I currently have spinning when loading wasm. I use that one for grid loading and its good there.

But I was thinking about something with more information to distract the user. Having some text come and go and some kind of progress bar going would be great.

![image](https://user-images.githubusercontent.com/2386572/148385619-5da60654-e6eb-42f5-8200-2cd0d8906540.png)

or 

![image](https://user-images.githubusercontent.com/2386572/148385733-ce2fc7e0-50b8-40c7-8ddc-4682765b1c7f.png)

![image](https://user-images.githubusercontent.com/2386572/148385917-e55100b5-3b8a-4de6-a6b7-844908001535.png)

or even just a round one with counting % (nice if "Progress" could be rolling text)
![image](https://user-images.githubusercontent.com/2386572/148386228-83fa7ead-7c2f-41df-b100-89c5d115cf8c.png)

Just something indicating progress and not infinite spin like with spinkit.

I at least think that there could be a room for improvement there. I will probably gobble something together and make it work ;-)

**Reply by stsrki** [reply]
It looks good. But it gets complicated. All those are probably native HTML elements(with some custom CSS), and they can work independently from Blazor. If we would create something similar as a Blazor component it would not work. Simply because the Blazor rendering engine must be up and running before any component can be shown.

**Reply by sturlath** [reply]
Yes I was afraid of that could be an issue. I´ll look into what I can do and if I end up with something interesting I´ll report back

Just found this cool collection of loaders if somebody is interested.. 
https://medium.muz.li/top-30-most-captivating-preloaders-for-your-website-95ed1beff99d

**Reply by David-Moreira** [reply]
Yea I would also not bother with trying to make a razor component work on the initial app load. (would probably need some gimmick with pre-render to just obtain the same result as plain html + css)... 
So yea... just get some cool looking html+css for your use case. 

If you want to do loading steps that gets more complicated... but you could probably just get an average of your app loading time (ofc it changes from user to user according to network speeds...) but let's imagine 5 seconds... and just display 
first 1s We are loading your assets...

next 2s Assets loaded...

next 2s Importing your stuff...

next 1s Almost there...

This would probably fool most users and give the sense that the app is doing smthg haha

**Reply by sturlath** [reply]
😅 that was what I was thinking... just that if the user thinks something important is happening he has no problem waiting 😉
I´ll look into some html+css and see if I cant fit it in..
